dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -534,20 +532,18 @@ public GHRepository getRepository(String repositoryId) {
     // The documentation doesn't list the possibilities https://github-api.kohsuke.org/apidocs/org/kohsuke/github/GHRef.GHObject.html#getType(),
     // but I'll assume it mirrors the 4 Git types: blobs, trees, commits, and tags.
     private String getCommitSHA(GHRef ref, GHRepository repository, String refName) throws IOException {
-        String sha = null;
-        if ("commit".equals(ref.getObject().getType())) {
-            sha = ref.getObject().getSha();
-        } else if (ref.getObject().getType().equals("tag")) {
-            GHTagObject tagObject = repository.getTagObject(sha);
-            sha = tagObject.getObject().getSha();
+        // Sha value if commit
+        String sha = ref.getObject().getSha();
+        if (ref.getObject().getType().equals("tag")) {
```

Comment:
Some minor things that might improve the readability and safety of this section:
1. Eliminate duplicate `ref.getObject().getType()` invocations.  Call once and store in local var.  Named `type`?
2. Use `Objects.equal` to compare strings, for graceful null handling.
3. Declare `sha` but don't immediately assign, set value in each conditional block of code.  That way, the compiler will let us know if one of the blocks didn't set a value, and it takes a bit less effort to figure out how it works, imho.

Reply:
> 2\. Use `Objects.equal` to compare strings, for graceful null handling.

Or put the string first, e.g., `"tag".equals...`.

But I think we had existing code like this, and I want to limit the scope of the change for the hotfix, just in case... So be careful...